### DISCUSSION TITLE
An alternative util function that can be used to reset recoil state

### DISCUSSION THREAD

**Discussion by kelvinsekx** [facebookexperimental/Recoil#1299] [Ideas]
Most atimes to reset a state requires an additional action other than a simple reset to initial state.
For example given a Timer app. Simply calling useResetRecoilState() would not stop the already called setInterval function. To do this, there is a need to reset the timerID too. See my example below to get a better glimpse.

`
const reset = () => {
  clearInterval(timeID);
  useResetRecoilState(timeState)
}

`<button onClick={reset}>Stop</button>`
`

Trying this out would obviously not work cause useResetRecoilState is a React hook that shouldnot be called in a callback...
Can we have an alternative util function that can be called in such manner in the reset function?

**Comment by drarmstr** [comment]
The `useResetRecoilState()` hook provides a callback which can in turn be called from a reset function.  Alternatively, hooks like `useRecoilCallback()` or `useRecoilTransaction_UNSTABLE()` can be used to produce callbacks that reset Recoil state as well.

**Comment by kelvinsekx** [comment]
Thanks that resolved it...passing a callback into the returned function from useResetRecoilState. I am only rewriting the example below so it could help any other person who tries reading this.

```javascript
// given this is react component

 const resetter = useRecoilState(atomName);
  <button onClick={()=>resetter(clearInterval(ID))}>Stop</button>
```
